docs/SCHEMA.md

```diff
@@ -0,0 +1,348 @@
+  }
+]
+```
+
+### `default`
+
+The recommended way to define a default value is to set it in the example variable declaration like so:
+
+```bash
+EXAMPLE=default-value
+```
+
+If for some reason you do not want to populate the value in that way, you can use the `# default:` comment.
+
+The `# default:` comment will override the value specified in the variable declaration if both a present.
+
+> Important: There won't be any parsing for the default value based on the format. The user of the parsed result is required to do so.
+
+**Default:** `null`
+
+**Example `.env` file:**
+
+```bash
+# description: Your own phone number
+MY_PHONE_NUMBER=+12345678901
+
+# description: Your Twilio number
+# default: +12223334444
+TWILIO_PHONE_NUMBER=
+```
+
+**Parsed Result (with omitted default values):**
+
+```json
+[
+  {
+    "key": "MY_PHONE_NUNMBER",
+    "description": "Your own phone number",
+    "default": "+12345678901"
+  },
+  {
+    "key": "TWILIO_PHONE_NUNMBER",
+    "description": "Your Twilio number",
+    "default": "+12223334444"
+  }
+]
+```
+
+### `configurable`
+
+If you want a variable to be copied over without the user being able to modify it, you can use the `configurable` option. It's important to note that the user could still change the value later on. Therefore you should still validate the value when using it.
+
+**Default:** `true`
+
+**Example `.env` file:**
+
+```bash
+# description: The path to the SMS webhook
+# format: url
+TWILIO_SMS_WEBHOOK_URL=/sms
+```
+
+**Parsed Result (with omitted default values):**
+
+```json
+[
+  {
+    "key": "TWILIO_SMS_WEBHOOK_URL",
+    "description": "Your own phone number",
+    "format": "url",
+    "default": "/sms"
+  }
+]
+```
+
+## Best Effort Guessing
+
+In order to be as compatible with existing `.env` files as possible, this schema will do some best effort guessing to determine meta information.
+
+This currently includes
+
+- `description`
+- `default`
+
+In all cases the special comments mentioned above will override the determined values
+
+**Example `.env` file:**
+
+```bash
+# Set these environment variables
+
+# Your Twilio Account SID
+TWILIO_ACCOUNT_SID=ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx
+
+# Your Twilio Auth Token
+# Get it in the Twilio Console
+TWILIO_AUTH_TOKEN=
+
+TWILIO_SMS_WEBHOOK_URL=
+```
+
+**Parsed Result (with omitted default values):**
+
+```json
+[
+  {
+    "key": "TWILIO_ACCOUNT_SID",
+    "description": "Your Twilio Account SID",
+    "default": "ACxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx"
+  },
+  {
+    "key": "TWILIO_AUTH_TOKEN",
+    "description": "Your Twilio Auth Token - Get it in the Twilio Console",
+    "default": null
+  },
+  {
+    "key": "TWILIO_SMS_WEBHOOK_URL",
+    "description": null,
+    "default": null
+  }
+]
+```
+
+Please note that multi-line comments will be merged into single lines using `-` as separator. UIs using this information might still decide to truncate the information. For long information consider using the `link:` comment.
```

Comment:
`-` as a separator seems like it might be problematic with legitimate uses of `-` within a single description line. What about an actual newline character or something like `\n`?

Reply:
I'm fine with using actual newlines. It will just mean that different UI's have to deal with it. Like this tool will probably convert it into a - because multiline for the CLI isn't support as far as I know. And for a browser UI you'd have to convert it into `<br>` but I'm fine with that.